lib/grape/validations/types/build_coercer.rb

```diff
@@ -54,8 +58,22 @@ def self.build_coercer(type, method = nil)
 
         # Virtus will infer coercion and validation rules
         # for many common ruby types.
-        Virtus::Attribute.build(conversion_type, converter_options)
+        to_cache(type, method, Virtus::Attribute.build(conversion_type, converter_options))
       end
+
+      def self.from_cache(type, method)
+        @virtus_cache[cache_key(type, method)]
+      end
+
+      def self.to_cache(type, method, value)
+        @virtus_cache[cache_key(type, method)] = value
+      end
+
+      def self.cache_key(type, method)
+        [type.to_s, method].flatten.join('_')
+      end
+
+      instance_variable_set(:@virtus_cache, {})
```

Comment:
For consistency I would either call this `@cache` or rename the method to read/write cache. Also what's the scope of this variable? Maybe it needs to be something like `__cache` or some other more internal variation?